upcloud/resource_upcloud_server_test.go

```diff
@@ -65,6 +78,204 @@ func TestUpcloudServer_changePlan(t *testing.T) {
+						"upcloud_server.my-server",
+						"network_interface.1.ip_address"),
+					testAccCheckServerIDNotEqual("upcloud_server.my-server", serverID),
+					testAccCheckNetwork("upcloud_server.my-server", 1, "upcloud_network.test_network_1"),
+					testAccGetServerID("upcloud_server.my-server", &serverID),
+				),
+			},
+			{
+				Config: testAccServerNetworkInterfaceConfig(
+					networkInterface{
+						niType: "utility",
+					},
+					networkInterface{
+						niType:     "private",
+						network:    true,
+						newNetwork: true,
+					},
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"upcloud_server.my-server", "plan", "2xCPU-4GB"),
+					resource.TestCheckResourceAttr(
+						"upcloud_server.my-server", "network_interface.#", "2"),
+					resource.TestCheckResourceAttr(
+						"upcloud_server.my-server",
+						"network_interface.0.type",
+						"utility"),
+					resource.TestCheckResourceAttr(
+						"upcloud_server.my-server",
+						"network_interface.0.ip_address_family",
+						"IPv4"),
+					resource.TestCheckResourceAttrSet(
+						"upcloud_server.my-server",
+						"network_interface.0.mac_address"),
+					resource.TestCheckResourceAttrSet(
+						"upcloud_server.my-server",
+						"network_interface.0.ip_address"),
+					resource.TestCheckResourceAttrSet(
+						"upcloud_server.my-server",
+						"network_interface.0.network"),
+					resource.TestCheckResourceAttr(
+						"upcloud_server.my-server",
+						"network_interface.1.type",
+						"private"),
+					resource.TestCheckResourceAttr(
+						"upcloud_server.my-server",
+						"network_interface.1.ip_address_family",
+						"IPv4"),
+					resource.TestCheckResourceAttrSet(
+						"upcloud_server.my-server",
+						"network_interface.1.mac_address"),
+					resource.TestCheckResourceAttrSet(
+						"upcloud_server.my-server",
+						"network_interface.1.ip_address"),
+					testAccCheckServerIDNotEqual("upcloud_server.my-server", serverID),
+					testAccCheckNetwork("upcloud_server.my-server", 1, "upcloud_network.test_network_11"),
+				),
+			},
+		},
+	})
+}
+
+func testAccGetServerID(resourceName string, id *string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		*id = s.RootModule().Resources[resourceName].Primary.ID
+
+		return nil
+	}
+}
+
+func testAccCheckServerIDNotEqual(resourceName string, id string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		newID := s.RootModule().Resources[resourceName].Primary.ID
+		if newID == id {
+			return fmt.Errorf("new server ID unexpectedly equals old ID: %s == %s", newID, id)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckNetwork(resourceName string, niIdx int, networkResourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		server := s.RootModule().Resources[resourceName]
+		network := s.RootModule().Resources[networkResourceName]
+		if network == nil {
+			fmt.Printf("%#v", s.RootModule().Resources)
```

Comment:
Is this a leftover or should we print at the other error states as well?

Reply:
Or maybe there is another reason for this?

Reply:
It's a bit of debug logging. The error returned should be enough to capture the test failure.